apps/chat/src/components/Chat/ChatInput/PromptVariablesDialog.tsx

```diff
@@ -84,18 +85,31 @@ export const PromptVariablesDialog: FC<Props> = ({
       );
 
       onSubmit(newContent);
-      onClose();
+      onClose && onClose();
     },
     [onClose, onSubmit, prompt.content, updatedVariables],
   );
 
+  const handleOnBlur = useCallback(
+    (index: number, e: FocusEvent<HTMLTextAreaElement>) => {
+      e.target.value = e.target.value.trim();
+      setUpdatedVariables((prev) => {
+        const updated = [...prev];
+        updated[index].value = e.target.value;
+        return [...updated];
```

Comment:
```suggestion
        updated[index].value = e.target.value;
        return updated;
```

we do not need to spread array which is not part of state